driver-core/src/main/java/com/datastax/driver/core/StatementFormatter.java

```diff
@@ -0,0 +1,1138 @@
+         * <p/>
+         * If the number of entries exceeds this threshold,
+         * printers should truncate it.
+         *
+         * @param maxOutgoingPayloadEntries the maximum number of printed outgoing payload entries.
+         * @throws IllegalArgumentException if the value is not > 0, or {@value StatementFormatterLimits#UNLIMITED} (unlimited).
+         * @return this (for method chaining).
+         */
+        public Builder withMaxOutgoingPayloadEntries(int maxOutgoingPayloadEntries) {
+            if (maxOutgoingPayloadEntries <= 0 && maxOutgoingPayloadEntries != StatementFormatterLimits.UNLIMITED)
+                throw new IllegalArgumentException("Invalid maxOutgoingPayloadEntries, should be > 0 or -1 (unlimited), got " + maxOutgoingPayloadEntries);
+            this.maxOutgoingPayloadEntries = maxOutgoingPayloadEntries;
+            return this;
+        }
+
+        /**
+         * Sets the maximum length, in bytes, allowed for a single outgoing payload value.
+         * The default is {@value DEFAULT_MAX_OUTGOING_PAYLOAD_VALUE_LENGTH}.
+         * <p/>
+         * If the payload value length in bytes exceeds this threshold,
+         * printers should truncate it.
+         *
+         * @param maxOutgoingPayloadValueLength the maximum length, in bytes, allowed for a single outgoing payload value.
+         * @throws IllegalArgumentException if the value is not > 0, or {@value StatementFormatterLimits#UNLIMITED} (unlimited).
+         * @return this (for method chaining).
+         */
+        public Builder withMaxOutgoingPayloadValueLength(int maxOutgoingPayloadValueLength) {
+            if (maxOutgoingPayloadValueLength <= 0 && maxOutgoingPayloadValueLength != StatementFormatterLimits.UNLIMITED)
+                throw new IllegalArgumentException("Invalid maxOutgoingPayloadValueLength, should be > 0 or -1 (unlimited), got " + maxOutgoingPayloadValueLength);
+            this.maxOutgoingPayloadValueLength = maxOutgoingPayloadValueLength;
+            return this;
+        }
+
+        /**
+         * Builds the {@link StatementFormatter} instance.
+         *
+         * @return the {@link StatementFormatter} instance.
+         */
+        public StatementFormatter build() {
+            StatementPrinterRegistry registry = new StatementPrinterRegistry();
+            registerDefaultPrinters(registry);
+            for (StatementPrinter<?> printer : printers) {
+                registry.register(printer);
+            }
+            StatementFormatterLimits limits = new StatementFormatterLimits(
+                    maxQueryStringLength, maxBoundValueLength, maxBoundValues, maxInnerStatements, maxOutgoingPayloadEntries, maxOutgoingPayloadValueLength);
+            return new StatementFormatter(registry, limits);
+        }
+
+        private void registerDefaultPrinters(StatementPrinterRegistry registry) {
+            registry.register(new DefaultStatementPrinter());
```

Comment:
> Do you think that is expected behavior?

It was, but I think your arguments are sound. Maybe we should make it so that custom printers have higher priority over bulilt-in ones.